File: src/main/java/proto/games/matchmaker/domain/player/PlayerService.java
package proto.games.matchmaker.domain.player;

import com.opencsv.CSVReader;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import proto.games.matchmaker.support.Converters;

import java.io.FileReader;
import java.net.URL;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

@Slf4j
@Service
public class PlayerService {
    public Set<Player> listAllPlayers() {
        List<List<String>> tuples = readFile();
        return tuples.stream()
                .map(this::createPlayer)
                .collect(Collectors.toSet());
    }

    private Player createPlayer(List<String> tuple){
        return Player.builder()
                .name(tuple.get(0))
                .email(tuple.get(1))
                .nickName(tuple.get(2))
                .rank(Converters.eloToRank(tuple.get(3)))
                .primaryPosition(Converters.rotaToPosition(tuple.get(4).split(",")[0]))
                .secondaryPosition(Converters.rotaToPosition(tuple.get(4).split(",")[1]))
                .build();
    }

    private List<List<String>> readFile() {
        try {
            return readCsv();
        } catch (Exception e) {
            log.error("Error to read csv {}", e.getMessage());
            throw new RuntimeException(e);
        }
    }

    private List<List<String>> readCsv() throws Exception {
        URL resource = getClass().getClassLoader().getResource("data/players.csv");

        List<List<String>> records = new ArrayList<>();
        try (CSVReader csvReader = new CSVReader(new FileReader(resource.getFile()));) {
            String[] values = null;
            while ((values = csvReader.readNext()) != null) {
                records.add(Arrays.asList(values));
            }
        }

        return records;
    }
}

File: src/main/kotlin/proto/games/matchmaker/domain/player/PlayerService.kt
package proto.games.matchmaker.domain.player

import com.opencsv.CSVReader
import org.springframework.stereotype.Service
import proto.games.matchmaker.support.Converters
import java.io.FileReader

@Service
class PlayerService {
    fun listAllPlayers(): Set<Player> {
        val tuples = readFile()
        return tuples
            .map { tuple -> createPlayer(tuple) }
            .toSet()
    }

    private fun createPlayer(tuple: Array<String>): Player {
        return Player(
            tuple[0],
            tuple[1],
            tuple[2],
            Converters.eloToRank(tuple[3]),
            Converters.rotaToPosition(tuple[4].split(",").toTypedArray()[0]),
            Converters.rotaToPosition(tuple[4].split(",").toTypedArray()[1])
        )
    }

    private fun readFile(): List<Array<String>> {
        return try {
            readCsv()
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }

    @Throws(Exception::class)
    private fun readCsv(): List<Array<String>> {
        val resource = javaClass.classLoader.getResource("data/players.csv")
        val records: List<Array<String>>;

        CSVReader(FileReader(resource.file)).use {
            records = it.iterator().asSequence().toList()
        }

        return records

    }
}